FeedAPIChallenge/Feed API/FeedImageMapper.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  Copyright © 2021 Essential Developer Ltd. All rights reserved.
+//
+
+import Foundation
+
+final class FeedImageMapper {
+	private static let HTTP_200 = 200
+
+	static func map(_ data: Data, from httpResponse: HTTPURLResponse) -> FeedLoader.Result {
+		guard httpResponse.statusCode == HTTP_200 else {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
+
+		return Result { try map(data) }
+	}
+
+	private static func map(_ data: Data) throws -> [FeedImage] {
+		guard let payload = try? JSONDecoder().decode(Payload.self, from: data) else {
+			throw RemoteFeedLoader.Error.invalidData
+		}
+		return payload.images.map { $0.feedImage }
```

Comment:
Suggestion: another way to use the `map` function is with key paths:
```suggestion
		return payload.images.map(\.feedImage)
```